glTF-BinExporter/RhinoDocGltfConverter.cs

```diff
@@ -0,0 +1,841 @@
+                    dracoGeoInfo.verticesMax = new float[] { point3f.X, point3f.Y, point3f.Z };
+
+                    // Triangle Stats
+                    dracoGeoInfo.trianglesNum = mesh.Faces.TriangleCount;
+                    dracoGeoInfo.trianglesMin = 0;
+                    dracoGeoInfo.trianglesMax = dracoGeoInfo.verticesNum - 1;
+
+                    // Normals Stats
+                    dracoGeoInfo.normalsNum = mesh.Normals.Count;
+                    vector3f = mesh.Normals.Min();
+                    dracoGeoInfo.normalsMin = new float[] { vector3f.X, vector3f.Y, vector3f.Z };
+                    vector3f = mesh.Normals.Max();
+                    dracoGeoInfo.normalsMax = new float[] { vector3f.X, vector3f.Y, vector3f.Z };
+
+                    // TexCoord Stats
+                    dracoGeoInfo.texCoordsNum = mesh.TextureCoordinates.Count;
+                    point2f = mesh.TextureCoordinates.Min();
+                    dracoGeoInfo.texCoordsMin = new float[] { point2f.X, point2f.Y };
+                    point2f = mesh.TextureCoordinates.Max();
+                    dracoGeoInfo.texCoordsMax = new float[] { point2f.X, point2f.Y };
+
+                    dracoGeoInfo.success = true;
+                }
+                geo.Dispose();
+                dracoCompression.Dispose();
+            }
+            finally
+            {
+                File.Delete(fileName);
+            }
+
+            return dracoGeoInfo;
+        }
+
+        private byte[] GetDracoBytes(string fileName)
+        {
+            using (FileStream stream = File.Open(fileName, FileMode.Open))
+            {
+                var bytes = new byte[stream.Length];
+                stream.Read(bytes, 0, (int)stream.Length);
+
+                return bytes;
+            }
+        }
+
+        public void WriteDracoBytes(byte[] bytes, out int bufferIndex, out int byteOffset)
+        {
+            if(options.UseBinary)
+            {
+                byteOffset = (int)binaryBufferStream.Position;
+                binaryBufferStream.Write(bytes, 0, bytes.Length);
+                bufferIndex = 0;
+            }
+            else
+            {
+                glTFLoader.Schema.Buffer buffer = new glTFLoader.Schema.Buffer()
+                {
+                    Uri = Constants.TextBufferHeader + Convert.ToBase64String(bytes),
+                    ByteLength = bytes.Length,
+                };
+                bufferIndex = dummy.Buffers.AddAndReturnIndex(buffer);
+                byteOffset = 0;
+            }
+        }
+
+        private void AddRhinoObjectBinary(Rhino.Geometry.Mesh[] rhinoMeshes, Rhino.DocObjects.Material material, Guid materialId, RhinoObject rhinoObject)
+        {
+            int materialIndex = GetMaterial(material, materialId);
+
+            var primitives = new List<MeshPrimitive>();
+
+            foreach (var rhinoMesh in rhinoMeshes)
+            {
+                rhinoMesh.Faces.ConvertQuadsToTriangles();
```

Comment:
I'm not sure we should remove quad check when serializing faces.
https://developer.rhino3d.com/api/RhinoCommon/html/M_Rhino_Geometry_Collections_MeshFaceList_ConvertQuadsToTriangles.htm
The quad to triangle conversion can fail. There's issues with tolerances that the Rhino Mesh takes into account when doing the split. While it's likely this will work 99% of the time we should keep the second check around. Or we can just remove this one and always do it our way always.

Reply:
YIKES! Alrighty. We'll keep it around for sure then.